### PR TITLE
update dependabot.yml

### LINKED ISSUE #2268: Dependabot config updates
## Summary
According to our [release policy](https://github.com/cosmos/gaia/blob/main/RELEASING.md), we only backport PRs if there are security issues. 

Therefore, for dependabot bumping deps version, we will adopt the following policy:
- dependabot makes PRs to update deps version against main
- only backport deps version bump to release branch if the bumped version has security fix.

## Proposal

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by yaruwangway**
closes: #2268

This PR udpates the dependabot.yml.
- update github-action packages with label auto-merge, PR against main branch
- update gomod packages, PR against main branch
- update gomod packages for security fix, PR against release/v9.0.x and release/v8.0.x

@mmulji-ic, please check [here](https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/configuring-dependabot-security-updates#managing-dependabot-security-updates-for-your-repositories) for admin to setup dependabot security updates for github repo. Thanks!

### CODE REVIEW

File: .github/dependabot.yml
Comment (mpoke) [inline .github/dependabot.yml:21]: What's does this label do?

**Review by mpoke — APPROVED** [review]
Nice

**Review by mpoke — APPROVED** [review]

File: .github/dependabot.yml
Comment (mmulji-ic) [inline .github/dependabot.yml]: Before we were not merging npm changes, some of the changes are problematic. 
Even if the job ran ok, the generated pages might not be ok. There are no sanity tests on the web stuff.
Would prefer we put this on hold till we migrate to docusaurus.

File: .github/dependabot.yml
Comment (mmulji-ic) [inline .github/dependabot.yml:25]: Is this criteria met? # Only allow automated security-related dependency updates on release branches.

File: .github/dependabot.yml
Comment (mmulji-ic) [inline .github/dependabot.yml:35]: Ditto here as above

File: .github/dependabot.yml
Comment (mpoke) [reply .github/dependabot.yml:25]: Yes, since `Dependabot security updates` is enabled. See https://github.com/cosmos/gaia/settings/security_analysis

File: .github/dependabot.yml
Comment (mpoke) [reply .github/dependabot.yml]: I removed the npm section completely.

**Review by mmulji-ic — APPROVED** [review]

**Pull request merged**
